integration-tests/transport/test_conservative_transport.py

```diff
@@ -0,0 +1,184 @@
+
+    # Equation
+    V = domain.spaces("HDiv")
+    eqn = CoupledTransportEquation(domain, active_tracers=tracers, Vu=V)
+    output = OutputParameters(dirname=dirname)
+
+    io = IO(domain, output)
+
+    # Set up the divergent, time-varying, velocity field
+    U = Lx/tmax
+    W = U/10.
+
+    def u_t(t):
+        xd = x - U*t
+        u = U - (W*pi*Lx/Hz)*cos(pi*t/tmax)*cos(2*pi*xd/Lx)*cos(pi*z/Hz)
+        w = 2*pi*W*cos(pi*t/tmax)*sin(2*pi*xd/Lx)*sin(pi*z/Hz)
+
+        u_expr = as_vector((u, w))
+
+        return u_expr
+
+    # Specify locations of the two Gaussians
+    xc1 = 5.*Lx/8.
+    zc1 = Hz/2.
+
+    xc2 = 3.*Lx/8.
+    zc2 = Hz/2.
+
+    def l2_dist(xc, zc):
+        return min_value(abs(x-xc), Lx-abs(x-xc))**2 + (z-zc)**2
+
+    lc = 2.*Lx/25.
+    m0 = 0.02
+
+    # Set the initial state from the configuration choice
+    if property == 'conservation':
+        f0 = 0.05
+
+        rho_t = 0.5
+        rho_b = 1.
+
+        rho_d_0 = rho_b + z*(rho_t-rho_b)/Hz
+
+        g1 = f0*exp(-l2_dist(xc1, zc1)/(lc**2))
+        g2 = f0*exp(-l2_dist(xc2, zc2)/(lc**2))
+
+        m_X_0 = m0 + g1 + g2
+    else:
+        f0 = 0.5
+        rho_b = 0.5
+
+        g1 = f0*exp(-l2_dist(xc1, zc1)/(lc**2))
+        g2 = f0*exp(-l2_dist(xc2, zc2)/(lc**2))
+
+        rho_d_0 = rho_b + g1 + g2
+
+        # Constant mass field
+        m_X_0 = m0 + 0*x
+
+    if space == 'diff_order_0':
+        VCG1 = FunctionSpace(mesh, 'CG', 1)
+        VDG1 = domain.spaces('DG1_equispaced')
+
+        suboptions = {'rho_d': RecoveryOptions(embedding_space=VDG1,
+                                               recovered_space=VCG1,
+                                               project_low_method='recover',
+                                               boundary_method=BoundaryMethod.taylor),
+                      'm_X': ConservativeRecoveryOptions(embedding_space=VDG1,
+                                                         recovered_space=VCG1,
+                                                         boundary_method=BoundaryMethod.taylor,
+                                                         rho_name='rho_d',
+                                                         orig_rho_space=V_rho)
+                      }
+    elif space == 'diff_order_1':
+        Vt_brok = FunctionSpace(mesh, BrokenElement(V_m_X.ufl_element()))
+        suboptions = {'rho_d': EmbeddedDGOptions(embedding_space=Vt_brok),
+                      'm_X': ConservativeEmbeddedDGOptions(rho_name='rho_d',
+                                                           orig_rho_space=V_rho)}
+    else:
+        suboptions = {}
+
+    opts = MixedFSOptions(suboptions=suboptions)
+
+    transport_scheme = SSPRK3(domain, options=opts, increment_form=False)
+    transport_methods = [DGUpwind(eqn, "m_X"), DGUpwind(eqn, "rho_d")]
+
+    # Timestepper
+    stepper = PrescribedTransport(eqn, transport_scheme, io, transport_methods, prescribed_transporting_velocity=u_t)

```

Comment:
Note that this will need updating when we merge the other branch up to head-of-trunk!